Updated the package versions and changed the arguments of set_id in examples/stand_alone/marker

DMaroo opened the pull request:
I changed the versions of the Ignition's transport package (4 -> 8) and msgs package (1 -> 5) in the CMakeLists.txt. Also changed the set_id arguments in the marker.cc example, so that the code does what it is intended to do. Tested it on my system, works fine for me. For more, check #2920.

DMaroo committed:
updated the versions of the required packages

DMaroo committed:
changed the argument of set_id

Changed the argument of set_id to a non-zero number, so as to prevent generating a new marker every time that particular marker gets changed. See https://github.com/osrf/gazebo/issues/2920 for more.

DMaroo commented:
@scpeters Could you please have a look at this, and review the PR? Thank you :)

j-rivero commented:
> I changed the versions of the Ignition's transport package (4 -> 8) and msgs package (1 -> 5) in the CMakeLists.txt.

Changing major versions of dependencies is a no-go in already released branches since it can potentially affect ABI, API or behavior which impacts negatively in stability.  Please revert the changes on them.

DMaroo committed:
reverted versions for compatibility

DMaroo commented:
@j-rivero Agreed! However, a mention of this in the example's associated README might be helpful. Shall I do that?

j-rivero commented:
> @j-rivero Agreed! However, a mention of this in the example's associated README might be helpful. Shall I do that?

Sounds good to me, thanks a lot Dhruv.

DMaroo committed:
Added description about editing CMakeLists.txt

Added dependencies. Also added a brief note to modify CMakeLists.txt to match the version of the dependencies present on one's machine.

DMaroo commented:
I have updated the README. Please consider merging this. Thank you.

DMaroo commented:
@j-rivero Could you please merge this?

chapulina commented:
> >    I changed the versions of the Ignition's transport package (4 -> 8) and msgs package (1 -> 5) in the CMakeLists.txt.

> Changing major versions of dependencies is a no-go in already released branches 

Actually in this case, changing the versions on that example would be a bugfix. The example is in the `gazebo11` branch and should use the correct dependencies so it can be used with Gazebo 11. I think the fix in https://github.com/osrf/gazebo/pull/3014/commits/8d09795d5dbcac249415768fd67189149ea5fa64 is correct.

DMaroo commented:
Yes. The reason why I updated the versions was because when I ran the examples, CMake was not able to find the outdated packages and the build process failed. It worked once I updated the dependency versions in `CMakeLists.txt`, so I thought it would be good if I added that as well to my PR. Should I revert https://github.com/osrf/gazebo/pull/3014/commits/a431a4e9e889cf103b11f8988da3e17750465297?

traversaro commented:
While debugging https://github.com/osrf/gazebo/issues/2766, I had to run the marker example, and I noticed that at least for Gazebo11 it was sufficient to just remove the lines related to `ignition-msgs1` and `ignition-transport4` at all, as those depending are already searched for once one executes `find_package(gazebo)` . See https://gist.github.com/traversaro/bac589712c2e14e0722777c6b61dc9cc#file-cmakelists-txt for an example of such simplified CMakeLists.txt that works fine in any case in compiling the marker example.

traversaro commented on examples/stand_alone/marker/marker.cc:
```diff
@@ -29,7 +29,7 @@ int main(int _argc, char **_argv)
   ignition::msgs::Marker markerMsg;
   markerMsg.set_ns("default");
-  markerMsg.set_id(0);
+  markerMsg.set_id(1);
```
Not directly related to this PR, but just based on my experience debugging https://github.com/osrf/gazebo/issues/2766 . Given that this example is the main documentation entry point on how to use the Gazebo marker feature, perhaps we could write in a comment here that the id 0 has a special meaning (to represet **not set id**, that result in a random id being generated and a new marker created every time) and so it should not be used?

DMaroo replied:
Agree. Given the sparse comments in the original code, I was hesitant to add many comments. However, I will add a short comment regarding the id.

DMaroo replied:
I have tried to add a note regarding set_id's behaviour. I hope that it is not too verbose. In case it is verbose, please feel free to suggest a better change. Thank you.

traversaro replied:
I think it is great! Probably I would just move it before the first set_id, so we explain to the readed why we assign id starting from 1 .

DMaroo commented:
@traversaro Yes. The build process succeeds even when we do not explicitly find the `ignition` packages in the `CMakeLists.txt`.

DMaroo committed:
Removed explicitly finding and linking ignition packages

For gazebo11, the packages `ignition-msgs` and `ignition-transport` do not need to be explicitly linked and included in the `CMakeLists.txt`. Just using the Gazebo flags, include directories and libraries is enough.

DMaroo committed:
Removed unnecesary information regarding dependencies

Related to a8f451b107ce3fb6edfb61d04b329fe00e11d196

traversaro commented:
> 
> 
> @traversaro Yes. The build process succeeds even when we do not explicitly find the `ignition` packages in the `CMakeLists.txt`.

Yes, in my case I found that useful as I was able to quickly compile the same example both against Gazebo9 and against Gazebo11 without any need for changing the CMakeLists.txt, so that seems to be a useful property in general.

traversaro commented on examples/stand_alone/marker/CMakeLists.txt:
```diff
@@ -6,12 +6,7 @@ link_directories(${GAZEBO_LIBRARY_DIRS})
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${IGNITION-TRANSPORT_CXX_FLAGS}")
-include_directories(${IGNITION-TRANSPORT_INCLUDE_DIRS})
-link_directories(${IGNITION-TRANSPORT_LIBRARY_DIRS})
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```
This line can probably be removed now?
```suggestion
```

DMaroo replied:
Thanks. Missed that line.

DMaroo committed:
Added comment on the behaviour of set_id(0)

DMaroo committed:
Removed redundant line

Co-authored-by: Silvio Traversaro <pegua1@gmail.com>

DMaroo committed:
Changed comment positioning

traversaro commented:
Thanks a lot @DhruvMaroo ! I think this is a great improvement and should mitigate problems such as https://github.com/osrf/gazebo/issues/2766 . I think we need to wait for some Gazebo mantainer to look at this now, but for me it is a great improvement.

DMaroo commented:
Could this be reviewed? Thanks.

adityapande-1995 commented:
ABI check (https://build.osrfoundation.org/job/gazebo-abichecker-any_to_any-ubuntu_auto-amd64/930/API_5fABI_20report/) is failing due to a removed symbol ``gazebo::rendering Distortion::Distort`` which seems to be unrelated to the changes made here. I'll run the example locally just to check once.

adityapande-1995 commented:
Tested locally, works fine !

j-rivero commented:
Sorry for the delay, changes looks good to me. Let's get fresh CI to be sure we are not breaking Brew. @osrf-jenkins run tests

scpeters commented:
CI failures are unrelated
